### PR TITLE
Ensure Metric watcher is always running

### PR DESCRIPTION
**Pull request by simathih**

### CODE REVIEW

**Review by hestolz — CHANGES_REQUESTED** [review]
as-is seems like this will error out, unless i'm missing something. did this pass testing?

File: AzureMonitorAgent/agent.py [hunk 1]
```diff
@@ -389,6 +389,10 @@ def enable():
     """
     global AMAServiceStartCommand, AMAServiceStatusCommand
 
+    # start the metrics process if its not already running
+    if (protected_settings is None or len(protected_settings) == 0) or (public_settings is not None and "proxy" in public_settings and "mode" in public_settings.get("proxy") and public_settings.get("proxy").get("mode") == "application"):
```
Comment (thperapp) [inline AzureMonitorAgent/agent.py:393]: nit: public_settings["proxy"] and public_settings["proxy"]["mode"] might be better for readability

[the remark this answers is not in the document]
Comment (simathih) [reply AzureMonitorAgent/agent.py:395]: fixed

**Review by hestolz — COMMENTED** [review]
if enable fails later on, will the metrics proc be left alive on the machine?

File: AzureMonitorAgent/agent.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (simathih) [reply AzureMonitorAgent/agent.py:395]: We have to test this once the build is done

**Review by hestolz — APPROVED** [review]

**Comment by simathih** [comment]
> if enable fails later on, will the metrics proc be left alive on the machine?

yeah, but this is by design...subsequent update would call disable() before removing the bad AMA install which would clean up the process

**Pull request merged**